airbyte-webapp/src/views/Connector/ConnectorForm/components/Property/Control.tsx

```diff
@@ -25,6 +25,18 @@ export const Control: React.FC<ControlProps> = ({ property, name, disabled, erro
   const [field, meta, helpers] = useField(name);
   const useDatepickerExperiment = useExperiment("connector.form.useDatepicker", true);
 
+  const onChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+      field.onChange(e);
+      if (!property.isRequired && e.target.value === "") {
```

Comment:
What's the reason for only doing this for optional fields? Could we not also set required fields to `undefined` if a user deletes the value?

Reply:
Given that an empty string will fail the yup validation anyway for required properties you are right that there is no difference in practice. Simplified the condition.